bezier-rs/lib/src/lib.rs

```diff
@@ -331,6 +331,90 @@ impl Bezier {
 		};
 		bezier_starting_at_t1.split(adjusted_t2)[t2_split_side]
 	}
+
+	/// Returns the closest point on the curve to the provided point.
+	/// Uses a searching algorithm akin to binary search that can be customized using the following parameters:
+	/// - `convergence_epsilon` - Difference used between floating point numbers to be considered as equal
+	/// - `convergence_limit` - Controls the number of iterations needed to consider that minimum distance to have converged
+	/// - `iteration_limit` - Controls the maximum total number of iterations to be used
+	pub fn project(&self, point: DVec2, convergence_epsilon: f64, convergence_limit: i32, iteration_limit: i32) -> DVec2 {
+		// First find the closest point from the results of a lookup table
+		let lut_size = 20;
+		let lut = self.compute_lookup_table(Some(lut_size));
+		let (minimum_position, minimum_distance) = utils::get_closest_point_in_lut(&lut, point);
+
+		// Get the t values to the left and right of the closest result in the lookup table
+		let mut left_t = (0.max(minimum_position - 1) as f64) / lut_size as f64;
+		let mut right_t = (lut_size.min(minimum_position + 1)) as f64 / lut_size as f64;
+
+		// Perform a finer search by finding closest t from 5 points between [left_t, right_t] inclusive
+		// Choose new left_t and right_t for a smaller range around the closest t and repeat the process
+		let mut final_t = left_t;
+		let mut distance;
+
+		// Increment minimum_distance to ensure that the distance < minimum_distance comparison will be true for at least one iteration
+		let mut new_minimum_distance = minimum_distance + 1.;
+		// Maintain the previous distance to identify convergence
+		let mut previous_distance;
+		// Counter to limit the number of iterations
+		let mut iteration_count = 0;
+		// Counter to identify how many iterations have had a similar result. Used for convergence test
+		let mut convergence_count = 0;
+		// Store calculated distances to minimize unnecessary recomputations
+		let mut distances: [f64; 5] = [
```

Comment:
can we replace 5 with a constant? that way later we can use `constant - 1` instead of 4 and avoid errors in the long run